main.go

```diff
@@ -370,7 +373,37 @@ func main() {
 		}
 	}
 	// handle profiler sink close on exit
-	defer prfl.Stop()
+	defer func() {
+		logger.Infow("closing profiler sink client before exiting", "sink", prfl.GetSinkName())
+		prfl.Stop()
+	}()
+
+	// tracer sink
+	ts, err := tracer.GetSink(logger, tracertypes.SinkDriver(cfg.Controller.TracerSink))
+	if err != nil {
+		logger.Errorw("error while creating profiler sink, switching to noop", "error", err)
+
+		ts, err = tracer.GetSink(logger, tracertypes.SinkDriverNoop)
+
+		if err != nil {
+			logger.Errorw("error while creating noop profiler sink", "error", err)
+		}
+	}
+	// handle tracer sink close on exit
+	defer func() {
+		logger.Infow("closing tracer sink client before exiting", "sink", ts.GetSinkName())
+
+		if err := ts.Stop(); err != nil {
+			logger.Errorw("error closing tracer sink client", "sink", ms.GetSinkName(), "error", err)
+		}
+	}()
+
+	// initiate Open Telemetry, set it up with the sink Provider
+	otel.SetTracerProvider(ts.GetProvider())
+
+	if ms.MetricRestart() != nil {
```

Comment:
Previously it was not done, what is this line for?

There is an error returned, retrieve it and log it please.
```suggestion
	if err := ms.MetricRestart()l; err != nil {
	  logger.Errorw("error sending MetricRestart", "sink", ms.GetSinkName(), "error" , err)
```

Reply:
It was done earlier in the file and was moved for readability. It sends a metric to notify the controller restart.